ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PostgresStorageBackend.scala

```diff
@@ -3,38 +3,18 @@
 
 package com.daml.platform.indexer.parallel
 
-import java.sql.{DriverManager, PreparedStatement, ResultSet}
+import java.sql.{Connection, PreparedStatement, ResultSet}
 
 import com.daml.lf.data.Ref
 import com.daml.ledger.participant.state.v1.Offset
 
 import scala.collection.mutable
-import scala.util.{Failure, Success, Try}
 
-trait PostgresDAO {
-  def insertBatch(batch: RawDBBatchPostgreSQLV1): Unit
+object PostgresStorageBackend extends StorageBackend[RawDBBatchPostgreSQLV1] {
 
-  def updateParams(ledgerEnd: Offset, eventSeqId: Long, configuration: Option[Array[Byte]]): Unit
-
-  def initialize: (Option[Offset], Option[Long])
-}
-
-case class JDBCPostgresDAO(jdbcUrl: String) extends PostgresDAO with AutoCloseable {
-
-  private val connection = {
-    val c = DriverManager.getConnection(jdbcUrl)
-    c.setAutoCommit(false)
-
-    val stmnt = c.createStatement()
-    stmnt.execute("SET synchronous_commit TO OFF;")
-    c.commit()
-    stmnt.close()
-
-    c
-  }
-
-  private val preparedInsertEventsBatchDivulgence = connection.prepareStatement(
-    """
+  private val preparedInsertEventsBatchDivulgence: Connection => PreparedStatement =
```

Comment:
I take your word for it that calling it repeatedly performs just as well as calling it once per session.

Reply:
well it will be slower per se, but let's hope it is not that much slower :crossed_fingers: 
we need the flow level benchmarks :pray: